.circleci/config.yml

```diff
@@ -488,6 +533,71 @@ jobs:
       - manage_api_cache:
           save: true
 
+  # Run Puppeteer e2e UI tests on deployed "test" environment.
+  puppeteer-e2e-test:
+    <<: *defaults
+    environment:
+      <<: *e2e_env
+      WORKBENCH_ENV: dev
+    working_directory: ~/workbench
+    parallelism: 3
+    steps:
+      - browser-tools/install-browser-tools
+      - checkout_init_git
+      - halt_job_noncode_changes
+      - run_e2e_test
+
+  # Deploy UI local server then run Puppeteer e2e tests on deployed "local" environment.
+  puppeteer-e2e-local-ui:
+    <<: *defaults
+    environment:
+      <<: *e2e_env
+      WORKBENCH_ENV: local
+    resource_class: medium+
+    working_directory: ~/workbench
+    parallelism: 1
+    steps:
+      - run:
+          name: Check if it is a pull request
```

Comment:
Better to just skip on master. This logic is not consistent with the other tests, and causes problems because it won't run again when a PR is opened.

Specifically consider the following PR flow:
- git push # this logic runs, skips since there's no PR yet
- create PR # nothing happens, Circle continues running the other steps as normal, puppeteer will show green since it skipped
- (may or may not happen, depending on PR) git push # follow-up commit, now it will run properly

Reply:
FWIW there's an option we can turn on in Circle to only kick off workflows for changes on branches that are in pull requests if that's desirable.

Reply:
any objection to turn on this flag?
<img width="1038" alt="Screen Shot 2020-05-19 at 3 15 03 PM" src="https://user-images.githubusercontent.com/35533885/82368551-9f739700-99e3-11ea-858c-39168a2d9a4e.png">
https://app.circleci.com/settings/project/github/all-of-us/workbench/advanced?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fall-of-us%2Fworkbench

Reply:
Interestingly, appears to already be on. I don't know who turned that on and when. So sounds like it should work as you expect.

Still, we're doing the same semantic check on this line, so it's good to just check master branch here for consistency as you've done: https://github.com/all-of-us/workbench/blob/8fc54652be3b1b322f0ea3e316b8d4455dd9de1b/.circleci/config.yml#L109

Reply:
oh, I turned it on after seeing Jay's comment. Honestly, I always thought it was on before.
For job `puppeteer-e2e-local-ui`, the command to skip job if it is on master branch is:
```
- run:
          name: Skip job if this is not a pull request
          command: |
            echo "PR: $CI_PULL_REQUEST"
            if [ ${CIRCLE_BRANCH} != "" ] && [ ${CIRCLE_BRANCH} == "master" ] ; then
              echo "Not a PR, skipping Puppeteer tests"
              circleci-agent step halt
            fi
```
Is that correct?
Another question:  How can `${CIRCLE_BRANCH} == ""`? I haven't seen that property with a empty string before.

Reply:
It can be empty when a tag is pushed, which happens during deploys.